cdc/processor/pipeline/sink.go

```diff
@@ -142,7 +142,71 @@ func (n *sinkNode) flushSink(ctx pipeline.NodeContext, resolvedTs model.Ts) (err
 }
 
 func (n *sinkNode) emitEvent(ctx pipeline.NodeContext, event *model.PolymorphicEvent) error {
-	n.eventBuffer = append(n.eventBuffer, event)
+	if event == nil || event.Row == nil {
+		return nil
+	}
+
+	colLen := len(event.Row.Columns)
+	preColLen := len(event.Row.PreColumns)
+	config := ctx.ChangefeedVars().Info.Config
+
+	// This indicates that it is an update event,
+	// and after enable old value internally by default(but disable in the configuration).
+	// We need to handle the update event to be compatible with the old format.
+	if !config.EnableOldValue && colLen != 0 && preColLen != 0 && colLen == preColLen {
+		handleKeyCount := 0
+		equivalentHandleKeyCount := 0
+		for i := range event.Row.Columns {
+			if event.Row.Columns[i].Flag.IsHandleKey() && event.Row.PreColumns[i].Flag.IsHandleKey() {
+				handleKeyCount++
+				colValueString := model.ColumnValueString(event.Row.Columns[i].Value)
+				preColValueString := model.ColumnValueString(event.Row.PreColumns[i].Value)
+				if colValueString == preColValueString {
+					equivalentHandleKeyCount++
+				}
+			}
+		}
+
+		// If the handle key columns are not updated, PreColumns is directly ignored.
+		if handleKeyCount == equivalentHandleKeyCount {
+			event.Row.PreColumns = nil
+			n.eventBuffer = append(n.eventBuffer, event)
```

Comment:
In original implement, if old-value is not enabled, and update on a row without handle key update, is there still two events, one delete and one replace?

Reply:
There is only one event and it will not be split. You can see the `old-cdc-update-non-index-col` in my test.